### PR TITLE
Resize frames in both dimensions

### PR DESCRIPTION
**Pull request by t-wissmann**
This implements a feature which was already indicated by the cursor shape:
resizing frames in both x and y direction. For this, it was necessary to
entirely change the drag handler for resizing frames.

As a little bonus, the special cursor shape persists during the drag.

The new test case essentially checks that resizing in both dimensions
works as expected. The existing testcase for 1-dimensional resize
could be simplified a lot using new attributes and the convenience
functions in the python bindings.

### CODE REVIEW

File: src/decoration.cpp [hunk 1]
```diff
@@ -249,6 +249,29 @@ ResizeAction Decoration::positionTriggersResize(Point2D p)
     return act;
 }
 
+/**
+ * @brief Find the most apropriate ResizeAction given the current
+ * cursor position. This is a very fuzzy version of positionTriggersResize()
+ * @param the cursor position
+ * @return
```
Comment (blarz) [inline src/decoration.cpp:256]: Add a description of the return value?

**Review by blarz — CHANGES_REQUESTED** [review]
Just two minor things, other than that it looks good me!

**Comment by t-wissmann** [comment]
Thanks! I've fixed the two instances.

File: src/decoration.cpp (hunk 1 above)
Comment (blarz) [reply src/decoration.cpp]: Thanks!

File: src/decoration.cpp [hunk 2]
```diff
@@ -249,6 +249,29 @@ ResizeAction Decoration::positionTriggersResize(Point2D p)
     return act;
 }
 
+/**
+ * @brief Find the most apropriate ResizeAction given the current
```
[the remark this answers is not in the document]
Comment (blarz) [reply src/decoration.cpp]: Thanks!

**Review by blarz — APPROVED** [review]

**Pull request merged**